shared/rakefiles/xk_util.rake

```diff
@@ -275,12 +278,16 @@ task :restore_snapshot_from_image_file, [:snapshot_files] => [@gcp_creds_file, :
     # database-storage-couchdb-couchdb-0-060619-195849 from the file name:
     # 2019-06-27_154922-pv-database-storage-couchdb-couchdb-0-060619-195849.tar.gz
     snapshot_name = snapshot_file[/database-storage-couchdb-couchdb-\d-\d+-\d+/, 0]
+    pv_zone = pv_zones[snapshot_name[/(([A-Za-z]+-)+[\d])/,0]]
     sh "#{@exekube_cmd} sh -c ' \
-      gcloud compute images create image-disk-pv-#{snapshot_name} --source-uri=#{snapshot_file}
-      gcloud compute disks create disk-pv-#{snapshot_name} --zone=#{pv_zones[snapshot_name[/(([A-Za-z]+-)+[\d])/,0]]} --image=image-disk-pv-#{snapshot_name}
-      gcloud compute disks snapshot disk-pv-#{snapshot_name} --zone=#{pv_zones[snapshot_name[/(([A-Za-z]+-)+[\d])/,0]]} --snapshot-names external-pv-#{snapshot_name}
-      gcloud -q compute images delete image-disk-pv-#{snapshot_name}
-      gcloud -q compute disks delete disk-pv-#{snapshot_name} --zone=#{pv_zones[snapshot_name[/(([A-Za-z]+-)+[\d])/,0]]}
+      function cleanup {
+        gcloud -q compute disks delete disk-pv-#{snapshot_name} --zone=#{pv_zone}
+        gcloud -q compute images delete image-disk-pv-#{snapshot_name}
+      }
+      gcloud compute images create image-disk-pv-#{snapshot_name} --source-uri=#{snapshot_file} || { cleanup; return 1; }
+      gcloud compute disks create disk-pv-#{snapshot_name} --zone=#{pv_zone} --image=image-disk-pv-#{snapshot_name} || { cleanup; return 1; }
```

Comment:
If this step fails, will `cleanup` be able to clean up the image created on the previous line? Or will L284 fail and prevent L285 from running to clean up the image?

Reply:
I added the `|| true` to the clean up commands just in the case that the first one fails